src/rax.c

```diff
@@ -1053,8 +1053,8 @@ int raxRemove(rax *rax, unsigned char *s, size_t len, void **old) {
         }
         if (child) {
             debugf("Unlinking child %p from parent %p\n", (void *)child, (void *)h);
-            raxNode *new = raxRemoveChild(h, child);
             size_t oldalloc = rax_alloc_size(h);
+            raxNode *new = raxRemoveChild(h, child);
```

Comment:
Here h (the parent), might be realloced depending on number of remaining childs. So we must query its size before. Note that child has already been freed, and the accompanying memory reduction accounted for. The pointer bits of child are only used to unlink it from h at this point.

Reply:
Good. This looks clear now with the `oldalloc` variable and the calculation on the next line `rax->alloc_size = rax->alloc_size - oldalloc + rax_alloc_size(new);`.